sdk/canton/community/participant/src/main/scala/com/digitalasset/canton/participant/admin/PackageUpgradeValidator.scala

```diff
@@ -185,8 +202,13 @@ class PackageUpgradeValidator(
       }
       .filter { case (_, version) => pkgVersion > version }
       .maxByOption { case (_, version) => version }
-      .traverse { case (pId, _) => lookupDar(pId) }
-      .map(_.flatten)
+      .traverse { case (pId, _) =>
+        lookupDar(pId, upgradingPackagesMap)(loggingContext).map(
+          _.getOrElse(
+            throw new IllegalStateException("failed to look up dar present in package map")
```

Comment:
In general we should log internal errors. 
A cheap way is to extend `com.digitalasset.daml.lf.InternalError`.

Reply:
Done